neighsyncd/neighsyncd.cpp

```diff
@@ -29,10 +32,23 @@ int main(int argc, char **argv)
             netlink.dumpRequest(RTM_GETNEIGH);
 
             s.addSelectable(&netlink);
+            if (sync.getRestartAssist()->isWarmStartInProgress())
+            {
+                sync.getRestartAssist()->readTableToMap();
+                sync.getRestartAssist()->startReconcileTimer(s);
+            }
             while (true)
             {
                 Selectable *temps;
                 s.select(&temps);
+                if (sync.getRestartAssist()->isWarmStartInProgress())
+                {
+                    if (sync.getRestartAssist()->checkReconcileTimer(temps))
+                    {
+                        sync.getRestartAssist()->reconcile();
+                        sync.getRestartAssist()->stopReconcileTimer(s);
```

Comment:
>stopReconcileTimer [](start = 49, length = 18)

I believe what you need is a oneshot selectable timer, and it is totally independent with RestartAssist. Just a refactoring suggestion here.

Reply:
Let's address this in later PR if needed, thx.